backend/store/postgres/queue.go

```diff
@@ -0,0 +1,92 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/sensu/sensu-go/backend/queue"
+)
+
+type Queue struct {
+	db           DBI
+	pollDuration time.Duration
+}
+
+func NewQueue(db DBI) *Queue {
+	return &Queue{
+		db:           db,
+		pollDuration: time.Second * 5,
+	}
+}
+
+// Enqueue a queue item
+func (q *Queue) Enqueue(ctx context.Context, item queue.Item) error {
+	_, err := q.db.Exec(ctx, queueEnqueue, item.Queue, item.Value)
+	return err
+}
+
+// Reserve reserves a queue item.
+// When the queue is empty Reserve will block and poll for new items until either
+// an item is found, or the context is cancelled.
+//
+// When Reserve returns a Reservation, the caller MUST Ack or Nack that Reservation.
+// Otherwise a transaction + connection could be leaked depending on the session
+// settings (i.e. idle_in_transaction_session_timeout.)
```

Comment:
This seems a bit dangerous if it's possible to reserve items faster than it is to deal with them. Thoughts?

Reply:
It is indeed a bit dangerous @echlebek.

For ad hoc scheduling it's a problem we wouldn't likely run into, as the plan is to have a single goroutine reserving and processing these items.

I could see us mitigating it by either adding a Capacity field to the Queue with some synchronized logic to limit open reservations, or by replacing the Reserve interface with a Handle method that accepts a callback and can be backed by a worker pool with a fixed size.